app/_features/room/contexts/connection-status-context.tsx

```diff
@@ -0,0 +1,46 @@
+// TODO : Add ability to Observe the ICE Connection & Connection Signal / Bandwidth
+
+import { room } from '@/_shared/utils/sdk';
+import { useContext, useEffect, useState, createContext } from 'react';
+
+const ConnectionContext = createContext({
+  sseConnection: false,
+});
+
+export const useConnectionContext = () => {
+  return useContext(ConnectionContext);
+};
+
+export function ConnectionProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [connectionState, setConnectionState] = useState(true);
+
+  useEffect(() => {
+    room.on(room.event.CHANNEL_OPENED, () => {
+      console.log('success subscribe to event endpoint');
+      setConnectionState(true);
+    });
+
+    room.on(room.event.CHANNEL_CLOSED, ({ reason }) => {
+      if (typeof reason != 'string')
+        console.log('failed to handle channel closed, reason type invalid');
+      if (reason === 'notfound') {
```

Comment:
Seems better to use this constant rather than hardcoded string

https://github.com/inlivedev/inlive-js-sdk/blob/455f73d55276ef78ac93f9af5578350ab370b489/packages/room/channel/channel.js#L3-L7

Reply:
the types wasn't exported before, now with the latest package the types have been exported, thankyou for review this PR